### DISCUSSION TITLE
how to define the return type for order_by?

### DISCUSSION THREAD

**Discussion by videni** [diesel-rs/diesel#3025] [Q&A]
I read [composing-application](https://diesel.rs/guides/composing-applications.html) and  [schema-in-depth](https://diesel.rs/guides/schema-in-depth.html) several times, still no idea to define the return type for order by.

```
pub fn by_customer<'a>(customer_id: Uuid) -> diesel::helper_types::Order<?, ?>
    {
        table
            .select(subscription_link_access::all_columns)
            .filter(columns::customer_id.eq(customer_id))
            .order_by(columns::subscriped_at.desc())
    }
```

```
pub type BoxedQuery<'a> = subscription_link_access::BoxedQuery<'a, Pg>;

```
I have tried the BoxedQuery, no lucky,  the BoxedQuery seems only suitable for select statement, not order by. please help.

**Comment by weiznich** [comment]
As this question is missing essential information like a minimal reproducible example I can only point to [the documentation](https://docs.diesel.rs/1.4.x/diesel/query_dsl/trait.QueryDsl.html#method.order_by) and state that your function will return the same type as `QueryDsl::order_by`.
